### ISSUE TITLE
push_args destination now requires full path with image name and tag when default transport is used

### ISSUE THREAD

**Issue by jakestec** [containers/ansible-podman-collections#830]
Hello,

I've been using the latest collection version (v1.15.4) with an existing playbook to build container images. When I went to push a new image today I got the following error related to the line of code below:

`Destination must be a full URL or path to a directory.`

https://github.com/containers/ansible-podman-collections/blob/552af3ef7ff62706ec4bd88e8f962e069c66eddc/plugins/modules/podman_image.py#L795

My task:
```
container_registry: "my.registry.domain.name"
...
    - name: Create and push EE => "{{ ee_name }}:{{ ee_tag }}"
      containers.podman.podman_image:
        name: "{{ ee_name }}"
        tag: "{{ ee_tag }}"
        path: "{{ ansible_env.PWD }}/build"
        push: true
        auth_file: "{{ lookup('ansible.builtin.env', 'XDG_RUNTIME_DIR') }}/containers/auth.json"
        validate_certs: true
        push_args:
          dest: "{{ container_registry }}"
```
Debugging I can see its not forming the push arguments correctly:
```
    "podman_actions": [
        "/usr/bin/podman image ls rcnet-ee:1.9 --format json",
        "/usr/bin/podman inspect rcnet-ee:1.9 --format json",
        "/usr/bin/podman push --tls-verify --authfile /run/user/1001/containers/auth.json my-ee:1.9 my.registry.domain.name",
```
We should see:
```
 "/usr/bin/podman push --tls-verify --authfile /run/user/1001/containers/auth.json my-ee:1.9 my.registry.domain.name/my-ee:1.9",
```

The above task is working in v1.12.1. I was able to fix the issue by modifying the destination in the task to include the image name and tag like so:

```
    - name: Create and push EE => "{{ ee_name }}:{{ ee_tag }}"
      containers.podman.podman_image:
        name: "{{ ee_name }}"
        tag: "{{ ee_tag }}"
        path: "{{ ansible_env.PWD }}/build"
        push: true
        auth_file: "{{ lookup('ansible.builtin.env', 'XDG_RUNTIME_DIR') }}/containers/auth.json"
        validate_certs: true
        push_args:
          dest: "{{ container_registry }}/{{ ee_name }}:{{ ee_tag }}"
```

I am not sure if this is intended, or an unplanned side effect of the changes implemented by https://github.com/containers/ansible-podman-collections/commit/4985d484155b2a0d6f90f21da99e6f152aad3d02 when doing some further checking on the transport type.

**Comment by sshnaidm** [comment]
It would help more if instead of putting variables names ` "{{ ee_name }}"` you put real values, hard to guess what you pass to the task.
If you pass `dest` as `my.registry.domain.name` only, it won't work, you need to set a name as well. The issue is that it worked before while it shouldn't have. The Podman itself fails if you pass the registry only in command line. We stick to Podman behavior now.
It's kind of side-effect of fixing other issues with transports and adding another features for pushing, we don't have a way to understand either it's a different image, registry, archive or anything else in `dest`, neither Podman does. That's why it's required now to put exactly what you need in `dest` (exactly as in Podman command line).

**Comment by jakestec** [comment]
Hey, thanks for the response and sorry for the delay I was away from my desk for a couple weeks.

The explanation makes sense, however maybe we should update the documentation to reflect that. Because from the current functionality this example from the docs wont work since the `dest` is just a registry and no name or tag is provided.

https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_image_module.html#examples
```
- name: Build and push an image using an auth file
  containers.podman.podman_image:
    name: nginx
    push: true
    auth_file: /etc/containers/auth.json
    push_args:
      dest: quay.io/acme
```

It should show:
```
- name: Build and push an image using an auth file
  containers.podman.podman_image:
    name: nginx
    push: true
    auth_file: /etc/containers/auth.json
    push_args:
      dest: quay.io/acme/nginx:tag
```

**Comment by sshnaidm** [comment]
@jakestec  Yeah, let's fix the docs and error message, would you like to propose a patch?
> Having to again specify the image name in dest is not obvious

Yes, it's not obvious if you have been doing this with previous versions of `podman_image`, but that won't work in Podman as well. So it's better to keep the consistency here.

**Comment by jakestec** [comment]
Sure thing, I've submitted my PR here: https://github.com/containers/ansible-podman-collections/pull/848

**Comment by sshnaidm** [comment]
Fixed in #848
